kbase-extension/static/kbase/js/widgets/appWidgets2/validation.js

```diff
@@ -453,11 +458,31 @@ define([
                 } else if (parsedValue.length > maxLength) {
                     diagnosis = Constants.DIAGNOSIS.INVALID;
                     errorMessage = 'the maximum length for this parameter is ' + maxLength;
-                } else if (regexp && !regexp.test(parsedValue)) {
-                    diagnosis = Constants.DIAGNOSIS.INVALID;
-                    errorMessage =
-                        'The text value did not match the regular expression constraint ' +
-                        constraints.regexp;
+                } else if (constraints.regexp) {
```

Comment:
So it turns out the previous regex checking code was (1) not used and (2) wrong anyway. This fixes that. The regex constraint was properly tested for in the `validators/text` module, but not in this one or the now-defunct `common/validation`. 

Incidentally, the `regex_constraint` app spec option isn't even documented, and the only apps that use it are testing ones. So this doesn't matter much anyway.